### PR TITLE
RB: add additional overriding hooks

### PR DESCRIPTION
**Pull request by pferreir**
This will be needed to add a "Statistics" pane in Burotel, with an icon inside the `SidebarMenu`.

### CODE REVIEW

File: indico/modules/rb/client/js/components/SidebarMenu.jsx [hunk 1]
```diff
@@ -165,6 +180,7 @@ export default connect(
     hasOwnedRooms: userSelectors.hasOwnedRooms(state),
   }),
   dispatch => ({
+    dispatchAction: dispatch,
```
Comment (pferreir) [inline indico/modules/rb/client/js/components/SidebarMenu.jsx]: If you have a better idea on how to do this (allow plugins to dispatch actions), you're welcome to share it!

File: indico/web/client/js/react/util/Overridable.jsx [hunk 2]
```diff
@@ -18,10 +18,10 @@ import {withRouter} from 'react-router-dom';
  * @param {Object|Function} extraProps - additional properties passed to the final component
  */
 export function parametrize(Component, extraProps) {
-  const ParametrizedComponent = ({...props}) => {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/util/Overridable.jsx]: This looks like it was meant to explicitly copy the props (or maybe it was just a leftover from when we had a non-...rest prop as well?)

Comment (pferreir) [reply indico/web/client/js/react/util/Overridable.jsx]: I didn't find any reason to actually copy the properties over there. That's why I removed it.

File: indico/modules/rb/client/js/components/SidebarMenu.jsx (hunk 1 above)
Comment (ThiefMaster) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: `const dispatch = useDispatch()` inside the plugin's component? not sure if importing stuff from react-redux in plugins works though... and i guess this type of redirection of calling hooks would be pretty nasty.

so probably your option is already the way to go there (i'd just keep the `dispatch` name and add a comment mentioning that we want to provide it to plugins so nobody removes it as being unused)

Comment (pferreir) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: The problem with the `dispatch` name is that along the way it got "stolen" by one of the HOCs. But I think we don't need to actually steal it (probably a remnant from some other thing).

Comment (pferreir) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: Added the comment too.

File: indico/modules/rb/client/js/components/SidebarMenu.jsx [hunk 3]
```diff
@@ -165,6 +180,9 @@ export default connect(
     hasOwnedRooms: userSelectors.hasOwnedRooms(state),
   }),
   dispatch => ({
+    // ATTENTION: this is **intentionally** passed as a prop, despite not being used,
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/components/SidebarMenu.jsx]: s/ATTENTION/XXX (most editors highlight FIXME, TODO and XXX in comments) ;)

Comment (pferreir) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: Mine doesn't 😉

Comment (pferreir) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: Isn't `XXX` meant for thing which is not quite ready?

Comment (ThiefMaster) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: looks like there might be an extension for it: microsoft/vscode#9899

Comment (ThiefMaster) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: I always interpreted it as "heads-up"

SO seems to agree: https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments

Comment (pferreir) [reply indico/modules/rb/client/js/components/SidebarMenu.jsx]: Other sources disagree, and to me it looks more like a FIXME or a TODO, so I'll leave "ATTENTION", which seems to be pretty self-explanatory. You can always configure your IDE to highlight those 😉

**Pull request merged**